packages/hdwallet-provider/src/index.ts

```diff
@@ -324,18 +325,19 @@ class HDWalletProvider {
       throw new Error("Mnemonic invalid or undefined");
     }
 
-    this.hdwallet = EthereumHDKey.fromMasterSeed(
-      Buffer.from(mnemonicToSeedSync(phrase, password))
+    const hdwallet = createAccountGeneratorFromSeedAndPath(
+      mnemonicToSeedSync(phrase, password),
+      this.walletHdpath.replace(/\/$/, "").split("/")
```

Comment:
So, this line makes me think that the function `createAccountGeneratorFromSeedAndPath` is badly abstracted.

If it took a string, and performed the `split("/")` itself, that would make sense.  If, by contrast, it simply took an array of number/boolean pairs, and didn't perform any string processing, that would also make sense.  As it is, though, the level of abstraction is somewhere inbetween, which makes no sense.  It *doesn't* perform the split, but it *does* check for the initial m/M, etc?

It would probably make more sense to break this up a bit.  Have the publicly exposed function in `@truflfe/hdwallet` take a string, not an array.  It could then perform the split, check for `m`, possibly perform other validation (currently there mostly isn't any, like I could pass in "m/1potato"), check for apostrophes, turn it all into a nice list of number/boolean pairs, and then pass that to another function that did the rest of the work.  That, to my mind, would make much more sense in terms of each function representing a sensible abstraction.

Reply:
This code is from ganache, where it makes a bit more sense. In any case, the idea is to move this package to ganache, where it originated, but we didn't want to wait a release cycle.

Ganache tries to avoid redundant validation for internal packages, so if validation is added here, we likely won't pull it back into Ganache. In other words: don't send these functions nonsense. 🤪

Reply:
(I do agree that it's a strange function)

Reply:
Oh, ha.  I assumed they were taking user input.  OK, do it your way then I guess!